liblist.c

```diff
@@ -116,11 +105,7 @@ int ListInsertEnd(List *l, void *key) {
     return 0;
 }
 
-/*
- * Remove o elemento do inicio da lista e o retorna em 'elemento'.
- * Retorna 1 se a operacao foi bem sucedida e 0 caso contrario.
- */
-void *ListRemoveStart(List *l) {
+void *ListSoftRemoveStart(List *l) {
     if (!ListIsEmpty(l)) {
         Node *target = l->head;
         l->head = target->next;
```

Comment:
Como vamos fazer o free(target->key) depois?? Aqui eu entendi que tu não fez o free porque está retornando target->key através do removed... Mas se você perder o ponteiro pra essa target->key fora dessa função pode dar b.o depois, nope??

Reply:
Target->key não precisa de free, cancela...

Reply:
Se o desejado for fazer o free logo após a remoção do nó, é possível fazê-lo diretamente no retorno da chamada por meio de free(ListSoftRemoveStart(lista)), sendo um papel de quem chamou. Se o desejado for fazer o free posteriormente, o papel de conservar e não perder o ponteiro também é de quem chamou. Se quem chamou perde ponteiros, o código está errado de qualquer maneira.

Reply:
Isssoo, aaah, suave... Se quem tá chamando a parada não perde ponteiro então tudo certo